### ISSUE TITLE
Merge all Genode base libs into one or two (was mantis id 121)

### ISSUE THREAD

**Issue by nfeske** [genodelabs/genode#18]
The set of functionality that we always link to Genode programs becomes pretty clear now. It consists of the libraries listed in 'base/mk/base-libs.mk'. To simplify the directory layout and make the distinction between base and non-base libraries more clear, those libraries could be merged into a single 'base' library. In contrast to other static libraries, this base library is guaranteed to be linked only once (to 'ld.lib.so' for dynamic binaries). Core is the only program that does not link to this library. It assembles the needed parts of the library by itself. Eventually, we could use two "base" libraries, namely 'cxx' and 'base' because the library description file for cxx is rather complicated.
